src/lib/configs/initConfig.ts

```diff
@@ -547,17 +547,7 @@ export function getConfigInitFunction<
     return {
       ...initializedReadonlyConfig,
       async $commit(): Promise<void> {
-        if (!initializedReadonlyConfig.$validateLatest(this)) {
-          throw new Error(
-            `Couldn't save config ${color.yellow(
-              configPath,
-            )}. ${await validationErrorToString(
-              initializedReadonlyConfig.$validateLatest.errors,
-            )}`,
-          );
-        }
-
-        const config = removeProperties({ ...this }, ([, v]) => {
+        const config = removeProperties({ ...(await this) }, ([, v]) => {
```

Comment:
Please explain this line for me.
If i understand this code correctly, `this` points out to the returned object.
Why we need `await` then?
Also it would be good to add a `type` for this here

Reply:
I mostly put it here to shut up typescript. It thinks this can be a promise because the function around it might return a promise. Doesn't really matter much if there is await here or not

Reply:
wow, looks fun!

can you put that to comment, plz?

Reply:
I suggest just type this value manually instead of putting comments or something
async $commit(): Promise<void> {
->
`async $commit(this: InitializedConfig<LatestConfig> | null): Promise<void> {`
It will make sure that `this` will never be `Promise` value and ts will tell you that if not.

Reply:
Eventually i made something like this
```
async $commit(this: InitializedConfig<LatestConfig>): Promise<void> {
        const config = removeProperties(this, ([, v]) => {
          return typeof v === "function";
        });
```

Reply:
I used your suggestion since it doesn't leave any useless code in runtime. What you suggest is essentially typecasting (but for some reason eslint doesn't consider this to be a typescasting even though it clearly is). Anyway I left a comment still because this typecasting requires at least some explanation. And you correctly noticed that there is no need for spread operator here since `removeProperties` creates a new object anyway